### PR TITLE
Implements accountExists method in XpringClient class

### PR DESCRIPTION
**Pull request by amiecorso**
## High Level Overview of Change
An accountExists method was created out of necessity in the xpring-wallet-system project, but belongs in the SDK.  

### Context of Change
This PR introduces the accountExists method to the XpringClientDecorator and all Xpring client classes, and implements unit tests.  Subsequent PRs will implement integration tests, and replace call sites for accountExists in the wallet-system with calls to the SDK.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)

## Before / After
accountExists now exists in Xpring-JS XpringClient.

## Test Plan
CI

- integration tests for accountExists
- wallet-system replacement of local accountExists calls to SDK calls.

### CODE REVIEW

File: test/default-xpring-client-test.ts [hunk 1]
```diff
@@ -401,4 +401,34 @@ describe('Default Xpring Client', function(): void {
       done()
     })
   })
+
+  it('Check if account exists - exists with valid address and positive balance', async function() {
```
Comment (keefertaylor) [inline test/default-xpring-client-test.ts]: I'd retitle these as what we actually care about: "- successful network request" and "- failing network request". Open to suggestions if you think this title is conveying useful information.

Comment (keefertaylor) [inline test/default-xpring-client-test.ts]: It would be cool to add an integration test for this too! See `test/integration-tests.ts`.

(You can do this in a separate PR if you'd like)

File: src/default-xpring-client.ts [hunk 2]
```diff
@@ -291,6 +291,29 @@ class DefaultXpringClient implements XpringClientDecorator {
...
+
+  /**
+   * Check if an address exists on the XRP Ledger.
+   *
+   * @param address The address to check the existence of.
+   * @returns A boolean if the account is on the blockchain.
+   */
+  public async accountExists(address: string): Promise<boolean> {
+    // TODO(keefertaylor): Xpring SDK should provide a way to check this functionality without using exception handling as a control flow mechanism.
+    // Checking for xAddress format also happens in getBalance itself... is there a cleaner way to use that result here instead of repeating this code?
```
[the remark this answers is not in the document]
Comment (amiecorso) [reply src/default-xpring-client.ts]: Cool, I just made myself another bootcamp asana task to do this.

File: src/default-xpring-client.ts [hunk 3]
```diff
@@ -291,6 +291,29 @@ class DefaultXpringClient implements XpringClientDecorator {
 
     return accountData
   }
+
+  /**
+   * Check if an address exists on the XRP Ledger.
+   *
+   * @param address The address to check the existence of.
+   * @returns A boolean if the account is on the blockchain.
```
[the remark this answers is not in the document]
Comment (amiecorso) [reply src/default-xpring-client.ts]: also replaced in XpringClient class

File: test/default-xpring-client-test.ts (hunk 1 above)
Comment (amiecorso) [reply test/default-xpring-client-test.ts]: will do in separate PR

File: test/legacy/legacy-default-xpring-client-test.ts [hunk 4]
```diff
@@ -453,4 +453,46 @@ describe('Legacy Default Xpring Client', function(): void {
       done()
     })
   })
+
+  it('Check if account exists - exists with valid address and positive balance', async function() {
+    // GIVEN a DefaultXpringClient.
+    const xpringClient = new LegacyDefaultXpringClient(
+      fakeSucceedingNetworkClient,
+    )
+    // WHEN the account does exist
```
Comment (keefertaylor) [inline test/legacy/legacy-default-xpring-client-test.ts]: can you apply that same vertical space here and line 474

File: test/default-xpring-client-test.ts [hunk 5]
```diff
@@ -401,4 +401,34 @@ describe('Default Xpring Client', function(): void {
...
+    // GIVEN a DefaultXpringClient.
+    const xpringClient = new DefaultXpringClient(fakeSucceedingNetworkClient)
```
[the remark this answers is not in the document]
Comment (keefertaylor) [reply test/default-xpring-client-test.ts]: Done in legacy-default-xpring-client-test but not here.

File: src/legacy/legacy-default-xpring-client.ts [hunk 6]
```diff
@@ -234,6 +234,21 @@ class LegacyDefaultXpringClient implements XpringClientDecorator {
...
+    // TODO: replace with refactored checkNotXAddress() when complete
+    const classicAddress = Utils.decodeXAddress(address)
+    if (!classicAddress) {
+      throw new Error(LegacyXpringClientErrorMessages.xAddressRequired)
+    }
+    try {
+      await this.getBalance(address)
+      return true
+    } catch (e) {
+      await this.getBalance(address)
```
Comment (keefertaylor) [inline src/legacy/legacy-default-xpring-client.ts]: This is a typo right? I think this will just throw another exception.

File: src/default-xpring-client.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (keefertaylor) [reply src/default-xpring-client.ts]: See conversation in legacy default client and apply that here.

File: src/legacy/legacy-default-xpring-client.ts [hunk 7]
```diff
@@ -234,6 +234,21 @@ class LegacyDefaultXpringClient implements XpringClientDecorator {
       return feeAmount
     })
   }
+
+  public async accountExists(address: string): Promise<boolean> {
+    // TODO: replace with refactored checkNotXAddress() when complete
```
Comment (keefertaylor) [inline src/legacy/legacy-default-xpring-client.ts]: Generally, prefer TODO's to have an owner so they don't get. 

So we could make this:
// TODO(amiecorso): Replace with refactored checkNotXAddress() when complete
// TODO(amiecorso): See asana.com/12345

Or we could drop it and just rely on the Asana task. 

All of these make me equally happy.

File: src/legacy/legacy-default-xpring-client.ts (hunk 6 above)
Comment (amiecorso) [reply src/legacy/legacy-default-xpring-client.ts]: oh yeah this totally is.  I removed this from default-xpring-client but forgot it here.  I think this was part of a linter weirdness experiment.

File: src/legacy/legacy-default-xpring-client.ts (hunk 7 above)
Comment (amiecorso) [reply src/legacy/legacy-default-xpring-client.ts]: i'll drop them because i've got the task!
